### ISSUE TITLE
List MooseEnum options in --dump

### ISSUE THREAD

**Issue by bwspenc** [idaholab/moose#12977]
## Rationale

We list the options for MooseEnum types on the webpage. We should similarly list them out in the output you get when you run a MOOSE app with the `--dump` option.

## Description

I don't really see a standardized way of writing the documentation strings for parameters with MooseEnum types. Sometimes, all of the options for the MooseEnum string are listed out, other times they are not. I think it's a bad idea to list them out in the doc string, because there's a potential for them to get out of sync.

In a recent discussion of this, it was suggested that we should list them out in the doc string so that they would show up in the `--dump` output. That seems like a horrible reason to list them out to me. Instead, we should just modify `--dump` to print out the enumeration options.

## Impact

This would involve adding additional output to what is provided by the `--dump` option, and probably doing some re-formatting of that output. Once that is done, we should also go through all of our documentation strings and make sure that they are not listing out the MooseEnum options.

**Comment by permcody** [comment]
How is this not already done? I thought we already had this! Oh well, it should be easy enough to add in.

**Comment by jmeziere** [comment]
@bwspenc @permcody Where would we find the code for the `--dump` output? It shouldn't take that long to add this in and reformat it.

**Comment by permcody** [comment]
outputs/formatters

**Comment by jmeziere** [comment]
Awesome, thanks. I'm seeing in [InputFileFormatter.C](https://github.com/idaholab/moose/blob/next/framework/src/outputs/formatters/InputFileFormatter.C) on line 144, the option for dump mode. Is getDocString what bwspenc was referring to for what we should not change? In other words, we shouldn't change that function, but add some new output beneath it somewhere? Also, what is meant by MooseEnum types? bwspenc mentioned that these are listed on the webpage somewhere. Would it be possible to get an example of that?

**Comment by permcody** [comment]
MooseEnum is a class in MOOSE (actually a few classes): https://mooseframework.inl.gov/docs/doxygen/moose/classMooseEnumBase.html

The purpose is to provide a "smart" enum type since C/C++ enums have several deficiencies. Our enum class can give you a list of valid items. It can also do case-agnostic comparisons, etc.

BTW - I think that `--dump` is falling further and further off the list of relevant things to work on. The original intent was to give a nice HIT format looking output of what input file syntax would look like. You use to be able to supply regular expressions as an argument to filter it down too. However, as MOOSE has evolved and we've changed the way syntax is printed some of those original features have just quit working. What is arguably more important is the `--json` formatter. That's the output that gets read into Peacock and used for creating actual input file syntax through the GUI interface. The `--json` formatter already has a list of MooseEnum options for each MooseEnum type. This has always been there and those options get used to build a "pick list" for MooseEnum types automatically. Should we add them to `--dump`? Maybe, but more work is needed with that format to get it useful enough for people to bother using it. I'm on the fence with what we should do here.

If you are going to go down the road of improving MooseEnum, or other things related to the help or GUI generation, there may be better tickets to look at:
https://github.com/idaholab/moose/issues/1774

Note: That we are also listing MooseEnum options on the MooseDocs page:
https://github.com/idaholab/moose/issues/9045

**Comment by kyriv-lab** [comment]
@GiudGiud can we close this issue now?

**Comment by GiudGiud** [comment]
closing as addressed by #33231
